eng/targets/Npm.Common.targets

```diff
@@ -56,7 +57,7 @@
   <Target Name="_Build"
     Condition="'$(IsBuildable)' != 'false'"
     Inputs="@(TSFiles)"
-    Outputs="$(BaseIntermediateOutputPath)\build-sentinel" >
+    Outputs="$(BaseIntermediateOutputPath)\build-sentinel;@(BuildOutputFiles)">
```

Comment:
Doubt this will do what you intend. MSBuild handles 1:1 relationships between inputs and outputs and m:1 but not m:m+1. Suggest removing build-sentinel from the outputs.

Reply:
It seems to work. My worry with removing it is that you'd have to require `BuildOutputFiles` to be non-empty. In the absence of this, MSBuild entirely skips the target since there are no outputs.

Reply:
What MSBuild actually does is compare the existence and timestamps of the inputs and outputs 1:1

Reply:
Right. But if the one-to-one mapping doesn't exist, msbuild just makes sure it's newer than all input files. That seems perfectly reasonable here. I did tweak this a bit, making specifying `BuildOutputFiles` optional by using a globbing pattern. It's still specified to determine if the directory was deleted after the project was built (that's what #13658 is all about) which effectively requires having an additional sentinel file.